src/core/Typeahead.js

```diff
@@ -261,6 +261,9 @@ export function getInitialState(props: Props): TypeaheadState {
 }
 
 export function clearTypeahead(state: TypeaheadState, props: Props) {
+  if (typeof props.onInputChange === "function") {
+    props.onInputChange("");
```

Comment:
As noted, `onInputChange` passes the change event as the second argument. It also doesn't seem like a good idea to trigger a callback during a state change. That should happen after the state change is complete.